ports/stm32/pyb_can.c

```diff
@@ -432,6 +432,20 @@ STATIC mp_obj_t pyb_can_send(size_t n_args, const mp_obj_t *pos_args, mp_map_t *
 
     HAL_StatusTypeDef status;
     #if MICROPY_HW_ENABLE_FDCAN
+    uint32_t timeout_ms = args[ARG_timeout].u_int;
+    uint32_t start = HAL_GetTick();
+    while (HAL_FDCAN_GetTxFifoFreeLevel(&self->can) == 0) {
+        if (timeout_ms == 0) {
+            mp_hal_raise(HAL_TIMEOUT);
```

Comment:
this is equivalent to `mp_raise_OSError(MP_ETIMEDOUT)` so would be better to just use that expression here

Reply:
fixed